axiom/nr_log_events.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nr_axiom.h"
+
+#include <math.h>
+#include <stdlib.h>
+
+#include "nr_log_events.h"
+#include "nr_analytics_events_private.h"
+#include "util_memory.h"
+
+bool nr_log_events_add_event(nr_analytics_events_t* events,
+                             const nr_log_event_t* event,
+                             nr_random_t* rnd) {
+  bool events_sampled = false;
+  char* log_event_json = NULL;
+  nr_analytics_event_t* log_event;
+
+  if (NULL == events) {
+    return false;
+  }
+  if (NULL == event) {
+    return false;
+  }
+
+  events_sampled = nr_analytics_events_is_sampling(events);
+
+  log_event_json = nr_log_event_to_json(event);
+  log_event = nr_analytics_event_create_from_string(log_event_json);
```

Comment:
Is it possible this returns a null or uninitialized pointer?  Should we check for null before adding the event?

Reply:
`nr_analytics_event_create_from_string` is basically an alias of `nr_strdup`. `nr_strdup` is `NULL` safe (handles `NULL` input gracefully) and will [exit](https://github.com/newrelic/newrelic-php-agent/blob/9fd43b56a296e761156db9aa08a1919a975cd56c/axiom/util_memory.c#L161) if it's not able to allocate memory rather than return `NULL`.